freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaInstanceHealthDetailsService.java

```diff
@@ -0,0 +1,161 @@
+        try (FreeIpaHealthCheckClient client = freeIpaHealthCheckClientFactory.getClient(stack, instance)) {
+            return client.nodeHealth();
+        } catch (FreeIpaClientException e) {
+            throw new RetryableFreeIpaClientException("Error during healthcheck", e);
+        } catch (Exception e) {
+            LOGGER.error("FreeIPA health check failed", e);
+            throw new RetryableFreeIpaClientException("FreeIPA health check failed", e);
+        }
+    }
+
+    private RPCResponse<Boolean> legacyFreeIpaHealthCheck(Stack stack, InstanceMetaData instance) throws FreeIpaClientException {
+        FreeIpaClient freeIpaClient = freeIpaClientFactory.getFreeIpaClientForStackForLegacyHealthCheck(stack, instance.getDiscoveryFQDN());
+        return freeIpaClient.serverConnCheck(freeIpaClient.getHostname(), instance.getDiscoveryFQDN());
+    }
+
+    private RPCResponse<Boolean> toBooleanRpcResponse(RPCResponse<CheckResult> nodeHealth) {
+        RPCResponse<Boolean> response = new RPCResponse<>();
+        response.setSummary(nodeHealth.getSummary());
+        response.setResult(isHealthCheckPassing(nodeHealth));
+        response.setCount(nodeHealth.getCount());
+        response.setTruncated(nodeHealth.getTruncated());
+        response.setMessages(nodeHealth.getMessages());
+        response.setCompleted(nodeHealth.getCompleted());
+        response.setFailed(nodeHealth.getFailed());
+        response.setValue(nodeHealth.getValue());
+        return response;
+    }
+
+    private NodeHealthDetails parseMessages(RPCResponse<CheckResult> rpcResponse, InstanceMetaData instanceMetaData) {
+        NodeHealthDetails nodeResponse = new NodeHealthDetails();
+        nodeResponse.setName(instanceMetaData.getDiscoveryFQDN());
+        nodeResponse.setInstanceId(instanceMetaData.getInstanceId());
+        if (isHealthCheckPassing(rpcResponse)) {
+            nodeResponse.setStatus(InstanceStatus.CREATED);
+        } else {
+            nodeResponse.setStatus(InstanceStatus.UNHEALTHY);
+            nodeResponse.setIssues(rpcResponse.getMessages().stream().map(RPCMessage::getMessage).collect(Collectors.toList()));
+        }
+        return nodeResponse;
+    }
+
+    private NodeHealthDetails legacyParseMessages(RPCResponse<Boolean> rpcResponse, InstanceMetaData instanceMetaData) {
+        String precedingMessage = MESSAGE_UNAVAILABLE;
+        NodeHealthDetails nodeResponse = new NodeHealthDetails();
+        nodeResponse.setStatus(rpcResponse.getResult() ? InstanceStatus.CREATED : InstanceStatus.UNHEALTHY);
+        nodeResponse.setName(instanceMetaData.getDiscoveryFQDN());
+        nodeResponse.setInstanceId(instanceMetaData.getInstanceId());
+        boolean found = false;
+        for (RPCMessage message : rpcResponse.getMessages()) {
+            Matcher nodeMatcher = NEW_NODE_PATTERN.matcher(message.getMessage());
+            if (nodeMatcher.find()) {
+                found = true;
+            }
+            if (!found) {
```

Comment:
_found_ variable is not necessary.  use  (!nodeMatcher.find()) should be good enough.